File: cpp/sorting/insertionSort.cpp
#include <vector>
#include <iostream>

using std::vector;
using std::cout;
using std::endl;

vector<int> insertionSort(vector<int>& arr) {
    for (int i = 1; i < arr.size(); i++) {
        int j = i - 1;
        while (j >= 0 && arr[j + 1] < arr[j]) {
            int tmp = arr[j + 1];
            arr[j + 1] = arr[j];
            arr[j] = tmp;
            j--;
        }
    }
    return arr;
}

int main() {
    std::vector<int> arr = {4, 3, 2, 1, 6};

    insertionSort(arr);

    for (auto n: arr) {
        cout << n << ' ';
    }
    cout << endl;
    return 0;
}

File: java/sorting/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    public static int[] insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int j = i - 1;
            while (j >= 0 && arr[j+1] < arr[j]) {
                int tmp = arr[j + 1];
                arr[j + 1] = arr[j];
                arr[j] = tmp;
                j--;
            }
        }
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {4, 3, 2, 1, 6};
        System.out.println(Arrays.toString(insertionSort(arr)));
    }
}
